zerver/views/realm.py

```diff
@@ -611,48 +597,62 @@ def update_realm_user_settings_defaults(
-    send_stream_typing_notifications: Optional[bool] = REQ(json_validator=check_bool, default=None),
-    send_private_typing_notifications: Optional[bool] = REQ(
-        json_validator=check_bool, default=None
-    ),
-    send_read_receipts: Optional[bool] = REQ(json_validator=check_bool, default=None),
-    user_list_style: Optional[int] = REQ(
-        json_validator=check_int_in(UserProfile.USER_LIST_STYLE_CHOICES), default=None
-    ),
-    email_address_visibility: Optional[int] = REQ(
-        json_validator=check_int_in(UserProfile.EMAIL_ADDRESS_VISIBILITY_TYPES), default=None
-    ),
+    message_content_in_email_notifications: Optional[Json[bool]] = None,
+    pm_content_in_desktop_notifications: Optional[Json[bool]] = None,
+    desktop_icon_count_display: Optional[
+        Json[
+            Annotated[
+                int, check_int_in_after_validator(UserProfile.DESKTOP_ICON_COUNT_DISPLAY_CHOICES)
+            ]
+        ]
+    ] = None,
+    realm_name_in_email_notifications_policy: Optional[
+        Json[
+            Annotated[
+                int,
+                check_int_in_after_validator(
+                    UserProfile.REALM_NAME_IN_EMAIL_NOTIFICATIONS_POLICY_CHOICES
+                ),
+            ]
+        ]
+    ] = None,
+    automatically_follow_topics_policy: Optional[
+        Json[
+            Annotated[
+                int,
+                check_int_in_after_validator(
+                    UserProfile.AUTOMATICALLY_CHANGE_VISIBILITY_POLICY_CHOICES
+                ),
+            ]
+        ]
+    ] = None,
+    automatically_unmute_topics_in_muted_streams_policy: Optional[
+        Json[
+            Annotated[
+                int,
+                check_int_in_after_validator(
+                    UserProfile.AUTOMATICALLY_CHANGE_VISIBILITY_POLICY_CHOICES
+                ),
+            ]
+        ]
+    ] = None,
+    automatically_follow_topics_where_mentioned: Optional[Json[bool]] = None,
+    presence_enabled: Optional[Json[bool]] = None,
+    enter_sends: Optional[Json[bool]] = None,
+    enable_drafts_synchronization: Optional[Json[bool]] = None,
+    email_notifications_batching_period_seconds: Optional[Json[int]] = None,
+    twenty_four_hour_time: Optional[Json[bool]] = None,
+    send_stream_typing_notifications: Optional[Json[bool]] = None,
+    send_private_typing_notifications: Optional[Json[bool]] = None,
+    send_read_receipts: Optional[Json[bool]] = None,
+    user_list_style: Optional[
+        Json[Annotated[int, check_int_in_after_validator(UserProfile.USER_LIST_STYLE_CHOICES)]]
+    ] = None,
+    email_address_visibility: Optional[
+        Json[
+            Annotated[int, check_int_in_after_validator(UserProfile.EMAIL_ADDRESS_VISIBILITY_TYPES)]
+        ]
+    ] = None,
 ) -> HttpResponse:
     if notification_sound is not None or email_notifications_batching_period_seconds is not None:
```

Comment:
Can you spit this file to a separate commit or PR from all the little files? This is a lot more mental effort to review than the rest of them combined, and it'd be nice to be able to merge the little file changes in a first pass.